Stop using a different WooCommerce version to run the WooCommerce HPOS tests [MAILPOET-4801]

rodrigoprimo opened the pull request:
## Description

See Jira ticket description for more details.

## Code review notes

_N/A_

## QA notes

I don't think this PR needs to be checked by QA.

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-4801]

## After-merge notes

_N/A_


[MAILPOET-4801]: https://mailpoet.atlassian.net/browse/MAILPOET-4801?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

rodrigoprimo committed:
Use latest WooCommerce version in the CircleCI jobs

[MAILPOET-4801]

rodrigoprimo committed:
Use latest WooCommerce version for Woo HPOS related tests

Before WooCommerce 7.1.0 was released, we were defining a specific
version of Woo that included HPOS to be used when running HPOS related
tests. Now that 7.1.0 was released and since we started using it by
default in bd51822b961a6f8ad10d3454cde912917acac0b6, we don't need to
specify a different Woo version for the HPOS tests.

[MAILPOET-4801]

samnajian reviewed (dismissed):
LGTM!
After fixing the merge conflicts can be merged to trunk directly.

rodrigoprimo commented:
Thanks, @samnajian. I fixed the conflicts and rebased the branch, but I'm not able to merge. The review you left is stale. Could you please check again and merge if everything is alright?
